dateutil/test/test_rrule.py

```diff
@@ -4614,6 +4614,14 @@ def testReplaceIfNotSet(self):
         self.assertEqual(list(newrr),
                              [datetime(1997, 1, 6)])
 
+    def testRaisesCorrectErrorForUnsupportedProperty(self):
+        rule = "X-ABC-MMSUBJ:19970902T090000"
+        assertRaisesRegex(self, ValueError, "unsupported property", rrulestr, rule, dtstart=datetime(1997, 9, 2, 9, 0))
+
+    def testRaisesCorrectErrorIfDatetimeContainsColons(self):
+        rule = "FREQ=YEARLY;WKST=MO;UNTIL=2019-03-29T00:59:59"
+        self.assertRaises(ValueError, rrulestr, rule, dtstart=datetime(1997, 9, 2, 9, 0))
```

Comment:
I don't know if there is a good test that isn't too tightly bound to the specific error message, but one problem with this test is that it wouldn't have failed before this.

Possibly the right move is to split the difference and say that in this case we'll expect the entire line to be shown, not just the specific property that failed:

```python
def testRaisesCorrectErrorIfDatetimeContainsColons(self):
    rule = "FREQ=YEARLY;WKST=MO;UNTIL=2019-03-29T00:59:59"
    with pytest.raises(ValueError, match=re.escape(rule)):
        rrulestr(rule)
```

Reply:
okay, will change accordingly